src/physics/lsm_driver.f90

```diff
@@ -797,9 +824,138 @@ subroutine lsm_init(domain,options)
 
             domain%canopy_water%data_2d = CQS2
             CQS2=0.01
-            where(domain%veg_type==ISWATER) domain%land_mask=kLC_WATER ! ensure VEGTYPE (land cover) and land-sea mask are consistent
+            ! where(domain%veg_type==ISWATER) domain%land_mask=kLC_WATER ! ensure VEGTYPE (land cover) and land-sea mask are consistent (BK 202208: this does not include lakes!!)
+            where((domain%veg_type==ISWATER) .OR. (domain%veg_type==ISLAKE)) domain%land_mask=kLC_WATER
+
+        endif
+
+        if(options%physics%watersurface==kWATER_LAKE) then
+        ! ____________ Lake model ______________________
+        ! From WRF's /run/README.namelist:  These could at some point become namelist options in ICAR?
+        ! lakedepth_default(max_dom)          = 50,      ! default lake depth (If there is no lake_depth information in the input data, then lake depth is assumed to be 50m)  
+        ! lake_min_elev(max_dom)              = 5,       ! minimum elevation of lakes. May be used to determine whether a water point is a lake in the absence of lake
+        !                                                  category. If the landuse type includes 'lake' (i.e. Modis_lake and USGS_LAKE), this variable is of no effects. 
+        ! use_lakedepth (max_dom)             = 1,       ! option to use lake depth data. Lake depth data is available from 3.6 geogrid program. If one didn't process
+        !                                                    the lake depth data, but this switch is set to 1, the program will stop and tell one to go back to geogrid
+        !                                                     program.
+        !                                     = 0, do not use lake depth data.
+
+            if (this_image()==1) write(*,*) "Initializing Lake model"
+
+            ! allocate arrays:  ( Could become separate subroutine)
```

Comment:
should / could some or all of these be located in the domain object as variable_t so they can be written to and read from restart files (and standard output files for that matter)